### PR TITLE
Tests for BetweenFactor

### PR DESCRIPTION
**Pull request by varunagrawal**
Added lots of tests which were missing, as well as uncommented existing tests so we have more confidence in this factor.

### CODE REVIEW

**Comment by varunagrawal** [comment]
Did this so we can be sure of the behavior of the PreintegratedContactFactors (which inherits from BetweenFactor) in GTDynamics. :slightly_smiling_face:

**Review by dellaert — COMMENTED** [review]
I'd approve but CI fails on many platforms :-) Maybe we uncovered a bug

**Comment by ProfFan** [comment]
```
118/255 Test #119: testBetweenFactor ......................Subprocess aborted***Exception:   0.26 sec
testBetweenFactor: /home/runner/work/gtsam/gtsam/gtsam/3rdparty/Eigen/Eigen/src/Core/PlainObjectBase.h:285: void Eigen::PlainObjectBase<Derived>::resize(Eigen::Index, Eigen::Index) [with Derived = Eigen::Matrix<double, 3, 1>; Eigen::Index = long int]: Assertion `(!(RowsAtCompileTime!=Dynamic) || (rows==RowsAtCompileTime)) && (!(ColsAtCompileTime!=Dynamic) || (cols==ColsAtCompileTime)) && (!(RowsAtCompileTime==Dynamic && MaxRowsAtCompileTime!=Dynamic) || (rows<=MaxRowsAtCompileTime)) && (!(ColsAtCompileTime==Dynamic && MaxColsAtCompileTime!=Dynamic) || (cols<=MaxColsAtCompileTime)) && rows>=0 && cols>=0 && "Invalid sizes when resizing a matrix or array."' failed.
```

**Comment by varunagrawal** [comment]
Yeah I saw the failures last night. This is a great thing!!

**Comment by varunagrawal** [comment]
It's interesting it is only failing on Debug builds, not Release builds. This is why I missed it locally, because I run Release builds to save time.

**Comment by varunagrawal** [comment]
Issue was a mistake in my test, put the wrong dimension.

**Review by dellaert — APPROVED** [review]

File: gtsam/slam/tests/testBetweenFactor.cpp [hunk 1]
```diff
@@ -103,14 +103,15 @@ TEST(BetweenFactor, Rot3Jacobians) {
...
-  Pose3 measured_value(Rot3(), Point3(1, 2, 3));
-  BetweenFactor<Pose3> factor(1, 2, measured_value, model);
+  Pose3 measured(Rot3(), Point3(1, 2, 3));
+  BetweenFactor<Pose3> factor(1, 2, measured, model);
 
+  Pose3 pose1, pose2(Rot3(), Point3(1, 2, 3));
   Values values;
-  values.insert(1, Pose3());
-  values.insert(2, Pose3(Rot3(), Point3(1, 2, 3)));
-  Vector3 error = factor.evaluateError(Pose3(), Pose3(Rot3(), Point3(1, 2, 3)));
```
Comment (ProfFan) [inline gtsam/slam/tests/testBetweenFactor.cpp]: Horror story that this actually works in Release...

**Pull request merged**